### PR TITLE
fix getCountEntities method added for Entity Count Query in DataStore

### PR DESCRIPTION
**Pull request by abhinav-qlogic**

### CODE REVIEW

File: google-cloud-clients/google-cloud-datastore/src/main/java/com/google/cloud/datastore/QueryResults.java [hunk 1]
```diff
@@ -70,4 +70,18 @@
...
+   * <p>A simple use case to count entities:
+   *
+   * <pre>{@code
+   * Query.newKeyQueryBuilder().setKind("Kind")
+   *                 .setOffset(Integer.MAX_VALUE).setLimit(0).build()
+   * QueryResults<Key> result = datasore.run(query);
+   * int count = results.getCountEntities();
+   * </pre>
+   */
+  int getCountEntities();
```
Comment (ajaaym) [inline google-cloud-clients/google-cloud-datastore/src/main/java/com/google/cloud/datastore/QueryResults.java]: why getCountEntities vs countEntities()?

**Review by ajaaym — CHANGES_REQUESTED** [review]

File: google-cloud-clients/google-cloud-datastore/src/main/java/com/google/cloud/datastore/QueryResults.java [hunk 2]
```diff
@@ -70,4 +70,18 @@
...
+   * Returns the number of entities count.
+   *
+   * <p>A simple use case to count entities:
+   *
+   * <pre>{@code
+   * Query.newKeyQueryBuilder().setKind("Kind")
+   *                 .setOffset(Integer.MAX_VALUE).setLimit(0).build()
+   * QueryResults<Key> result = datasore.run(query);
+   * int count = results.getCountEntities();
+   * </pre>
```
Comment (rahulKQL) [inline google-cloud-clients/google-cloud-datastore/src/main/java/com/google/cloud/datastore/QueryResults.java]: Please replace `</pre>`  with  ` }</pre>`

File: google-cloud-clients/google-cloud-datastore/src/main/java/com/google/cloud/datastore/QueryResultsImpl.java [hunk 3]
```diff
@@ -124,4 +125,22 @@ public int getSkippedResults() {
...
+  @Override
+  public int getCountEntities() {
+    return getEntitiesCount();
+  }
+
+  private int getEntitiesCount() {
+    int count = 0;
+    Integer limit = ((KeyQuery) this.query).getLimit();
+    if (null != limit && limit == 0) {
+      count = runQueryResponsePb.getBatch().getSkippedResults();
```
Comment (rahulKQL) [inline google-cloud-clients/google-cloud-datastore/src/main/java/com/google/cloud/datastore/QueryResultsImpl.java]: Please extract `runQueryResponsePb.getBatch()` into a local variable

**Review by rahulKQL — CHANGES_REQUESTED** [review]

File: google-cloud-clients/google-cloud-datastore/src/main/java/com/google/cloud/datastore/QueryResults.java (hunk 1 above)
Comment (abhinav-qlogic) [reply google-cloud-clients/google-cloud-datastore/src/main/java/com/google/cloud/datastore/QueryResults.java]: done

File: google-cloud-clients/google-cloud-datastore/src/main/java/com/google/cloud/datastore/QueryResults.java (hunk 2 above)
Comment (abhinav-qlogic) [reply google-cloud-clients/google-cloud-datastore/src/main/java/com/google/cloud/datastore/QueryResults.java]: done

File: google-cloud-clients/google-cloud-datastore/src/main/java/com/google/cloud/datastore/QueryResultsImpl.java (hunk 3 above)
Comment (abhinav-qlogic) [reply google-cloud-clients/google-cloud-datastore/src/main/java/com/google/cloud/datastore/QueryResultsImpl.java]: done

**Pull request closed without merging**